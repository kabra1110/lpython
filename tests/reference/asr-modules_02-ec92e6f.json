{
    "basename": "asr-modules_02-ec92e6f",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/modules_02.py",
    "infile_hash": "c3ce0b2b9780f27f787297f75e5477e990481b962dcee420809540e0",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-modules_02-ec92e6f.stdout",
    "stdout_hash": "51c58e43d231fdb96250975fb6064062d7855cb9da9fad762681ad8d",
    "stderr": "asr-modules_02-ec92e6f.stderr",
    "stderr_hash": "132af04271d3bfd523848990e734bfa3c0aed6e4b85ec4eb87e66720",
    "returncode": 0
}