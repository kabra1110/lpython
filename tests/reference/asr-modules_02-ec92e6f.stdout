(TranslationUnit
    (SymbolTable
        1
        {
            _global_symbols:
                (Module
                    (SymbolTable
                        9
                        {
                            _lpython_main_program:
                                (Function
                                    (SymbolTable
                                        8
                                        {
                                            
                                        })
                                    _lpython_main_program
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    [main0]
                                    []
                                    [(SubroutineCall
                                        9 main0
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            f:
                                (ExternalSymbol
                                    9
                                    f
                                    2 f
                                    modules_02b
                                    []
                                    f
                                    Public
                                ),
                            main0:
                                (Function
                                    (SymbolTable
                                        6
                                        {
                                            x:
                                                (Variable
                                                    6
                                                    x
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    main0
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    [f]
                                    []
                                    [(=
                                        (Var 6 x)
                                        (IntegerBinOp
                                            (IntegerBinOp
                                                (IntegerConstant 2 (Integer 4))
                                                Add
                                                (IntegerConstant 3 (Integer 4))
                                                (Integer 4)
                                                (IntegerConstant 5 (Integer 4))
                                            )
                                            Mul
                                            (IntegerConstant 5 (Integer 4))
                                            (Integer 4)
                                            (IntegerConstant 25 (Integer 4))
                                        )
                                        ()
                                    )
                                    (Assert
                                        (IntegerCompare
                                            (Var 6 x)
                                            Eq
                                            (IntegerConstant 25 (Integer 4))
                                            (Logical 4)
                                            ()
                                        )
                                        ()
                                    )
                                    (SubroutineCall
                                        9 f
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    _global_symbols
                    [modules_02b]
                    .false.
                    .false.
                ),
            main_program:
                (Program
                    (SymbolTable
                        7
                        {
                            _lpython_main_program:
                                (ExternalSymbol
                                    7
                                    _lpython_main_program
                                    9 _lpython_main_program
                                    _global_symbols
                                    []
                                    _lpython_main_program
                                    Public
                                )
                        })
                    main_program
                    [_global_symbols]
                    [(SubroutineCall
                        7 _lpython_main_program
                        ()
                        []
                        ()
                    )]
                ),
            modules_02b:
                (Module
                    (SymbolTable
                        2
                        {
                            f:
                                (Function
                                    (SymbolTable
                                        5
                                        {
                                            
                                        })
                                    f
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    [g]
                                    []
                                    [(SubroutineCall
                                        2 g
                                        ()
                                        []
                                        ()
                                    )
                                    (Print
                                        ()
                                        [(StringConstant
                                            "f()"
                                            (Character 1 3 ())
                                        )]
                                        ()
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            g:
                                (ExternalSymbol
                                    2
                                    g
                                    3 g
                                    modules_02c
                                    []
                                    g
                                    Public
                                )
                        })
                    modules_02b
                    [modules_02c]
                    .false.
                    .false.
                ),
            modules_02c:
                (Module
                    (SymbolTable
                        3
                        {
                            g:
                                (Function
                                    (SymbolTable
                                        4
                                        {
                                            
                                        })
                                    g
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    [(Print
                                        ()
                                        [(StringConstant
                                            "g()"
                                            (Character 1 3 ())
                                        )]
                                        ()
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    modules_02c
                    []
                    .false.
                    .false.
                )
        })
    []
)
